src/python_testing/TC_FAN_3_5.py

```diff
@@ -21,207 +21,1090 @@
+    def desc_TC_FAN_3_5(self) -> str:
+        return "[TC-FAN-3.5] Optional step functionality with DUT as Server"
+
+    def steps_TC_FAN_3_5(self):
+        return [TestStep(1, "[FC] Commissioning already done.", is_commissioning=True),
+                TestStep(2, "[FC] TH checks the DUT for the presence of the OnOff cluster.",
+                         "If the cluster is present, set it to On."),
+                TestStep(3, "[FC] TH reads from the DUT the SpeedMax attribute.",
+                         "Store value for future reference."),
+                TestStep(4, "[FC] TH reads from the DUT the FanModeSequence attribute.",
+                         "Store value for future reference."),
+                TestStep(5, "[FC] TH determines the PercentSetting range per Step command.",
+                         "- Set PercentSetting to 0. - Send Step command with direction=Increase, lowestOff=True, and wrap=False. - Read the resulting PercentSetting attribute report value. - Calculate the PercentSetting range per Step command. - Store value for future reference."),
+
+                # LowestOff tests
+                TestStep(6, "[FC] TH tests Step command behavior of the LowestOff field.",
+                         "- Setup a Step command with: -- Direction=Decrease -- Wrap=False -- LowestOff=True."),
+                TestStep(7, "[FC] Initialize the PercentSetting attribute to 100.",
+                         "- Verify that the SpeedSetting attribute value is set to SpeedMax. - Verify that the FanMode attribute value is set to High."),
+                TestStep(8, "[FC] Subscribe to PercentSetting, SpeedSetting, and FanMode attributes.",
+                         "Listen for attribute report updates."),
+                TestStep(9, "[FC] TH sends Step commands iteratively.",
+                         "- Verify that the PercentSetting attribute value reaches 0. - When the PercentSetting attribute value reaches 0, send an additional Step command to veiryf that the PercentSetting attribute value stays at 0."),
+                TestStep(10, "[FC] Read the resulting attribute reports from each subscription after the Step commands.",
+                         "- Verify that the attribute report values from each subscription are in descending order. - If the number of PercentSetting reports is greater than the number of FanMode reports: -- Verify that all the expected FanMode values are present in the reports in accordance with the FanModeSequence attribute value. - If the number of PercentSetting reports is greater or equal than the number of SpeedSetting reports: -- Verify that all the expected SpeedSetting values are present in the reports in accordance with the SpeedMax attribute value. - Save the resulting descending attribute report values from each subscription as a baseline for comparison with the ascending values from the next steps."),
+
+                TestStep(11, "[FC] TH tests Step command behavior of the LowestOff field.",
```

Comment:
This isn't verifying lowest off though - it's verifying that the step works the same increasing and decreasing. If this were verifying the other behaviour of lowest off, wouldn't you go from 100 down to whatever was just above 0 and verify that it didn't turn off?